File: Popcorn4/app/src/main/java/com/pandey/popcorn4/fvrtmovies/data/FvrtMovieDbObjectConverter.java
package com.pandey.popcorn4.fvrtmovies.data;

import androidx.annotation.NonNull;

import com.pandey.popcorn4.db.FvrtMoviesDbObject;
import com.pandey.popcorn4.movie.data.MovieInfo;

public class FvrtMovieDbObjectConverter {

    public static FvrtMoviesDbObject toDbObject(@NonNull MovieInfo movieInfo) {
        movieInfo.getMovieLanguage();
        movieInfo.getMovieTitle();
        return new FvrtMoviesDbObject(
                movieInfo.getMovieId(),
                movieInfo.getMoviePoster(),
                movieInfo.getMovieTitle(),
                movieInfo.getMovieMovieDescription(),
                movieInfo.getMovieStar(),
                movieInfo.getMovieLanguage(),
                System.currentTimeMillis()
        );
    }

    public static MovieInfo fromDbObject(FvrtMoviesDbObject fvrtMoviesDbObject) {
        return new MovieInfo() {
            @Override
            public int getMovieId() {
                return fvrtMoviesDbObject.movieId;
            }

            @NonNull
            @Override
            public String getMoviePoster() {
                return fvrtMoviesDbObject.moviePosterPath;
            }

            @NonNull
            @Override
            public String getMovieTitle() {
                return fvrtMoviesDbObject.movieTitle;
            }

            @Override
            public int getMovieStar() {
                return fvrtMoviesDbObject.movieStarCount;
            }

            @NonNull
            @Override
            public String getMovieLanguage() {
                return fvrtMoviesDbObject.movieLanguage;
            }

            @NonNull
            @Override
            public String getMovieMovieDescription() {
                return fvrtMoviesDbObject.movieDescription;
            }
        };
    }

}

File: Popcorn4/app/src/main/java/com/pandey/popcorn4/data/network/db/FvrtMovieDbObjectConverter.kt
package com.pandey.popcorn4.data.network.db

import com.pandey.popcorn4.movie.data.MovieInfo

class FvrtMovieDbObjectConverter {

    companion object {

        fun toDbObject(movieInfo: MovieInfo): FvrtMoviesDbObject? {
            movieInfo.movieLanguage
            movieInfo.movieTitle
            return FvrtMoviesDbObject(
                    movieInfo.movieId,
                    movieInfo.moviePoster!!,
                    movieInfo.movieTitle,
                    movieInfo.movieMovieDescription!!,
                    movieInfo.movieStar,
                    movieInfo.movieLanguage!!,
                    System.currentTimeMillis()
            )
        }

        fun fromDbObject(fvrtMoviesDbObject: FvrtMoviesDbObject): MovieInfo? {
            return object : MovieInfo {
                override fun getMovieId(): Int {
                    return fvrtMoviesDbObject.movieId
                }

                override fun getMoviePoster(): String {
                    return fvrtMoviesDbObject.moviePosterPath
                }

                override fun getMovieTitle(): String {
                    return fvrtMoviesDbObject.movieTitle!!
                }

                override fun getMovieStar(): Int {
                    return fvrtMoviesDbObject.movieStarCount
                }

                override fun getMovieLanguage(): String {
                    return fvrtMoviesDbObject.movieLanguage
                }

                override fun getMovieMovieDescription(): String {
                    return fvrtMoviesDbObject.movieDescription
                }
            }
        }
    }
}